src/goto-programs/rebuild_goto_start_function.cpp

```diff
@@ -69,6 +64,21 @@ bool rebuild_goto_start_function_baset<maybe_lazy_goto_modelt>::operator()()
   return return_code;
 }
 
+std::unique_ptr<languaget> get_entry_point_language(
+  const symbol_table_baset &symbol_table,
+  const optionst &options,
+  message_handlert &message_handler)
+{
+  const irep_idt &mode = get_entry_point_mode(symbol_table);
+
+  // Get the relevant languaget to generate the new entry point with
+  std::unique_ptr<languaget> language = get_language_from_mode(mode);
+  INVARIANT(language, "No language found for mode: " + id2string(mode));
```

Comment:
In which cases would we not get a language back?

Reply:
For example, when this is called in a context where the language hasn't been registered.

Reply:
Still not sure when this would happen.

Reply:
E.g. CBMC doesn't register Java: https://github.com/diffblue/cbmc/blob/develop/src/cbmc/cbmc_languages.cpp#L27